lib/fiddle/ffi_backend.rb

```diff
@@ -189,7 +189,10 @@ def initialize(ret, args, abi = Function::DEFAULT)
       return_type = Fiddle::FFIBackend.to_ffi_type(@ctype)
       raise "#{self.class} must implement #call" unless respond_to?(:call)
       callable = method(:call)
-      @function = FFI::Function.new(return_type, ffi_args, callable, convention: abi)
+      wrapper = lambda do |*args|
+        callable.(*args.map { |v| v.is_a?(FFI::Pointer) ? Pointer.new(v) : v })
```

Comment:
Can we use `call` directly here?

```suggestion
        call(*args.map { |v| v.is_a?(FFI::Pointer) ? Pointer.new(v) : v })
```